### PR TITLE
Adds support for writing Capybara specs

### PR DESCRIPTION
**Pull request by TildeWill**
- uses headless browser via phantomjs/poltergeist
- includes an example spec for creating an Agent
- includes a helper for interacting with Select2 boxes

Additionally, having Capybara will be helpful for my work with Omniscope as I add specs to give me confidence users can manage their billing flows correctly.

### CODE REVIEW

**Comment by cantino** [comment]
Nice!  This is a good improvement.  Let me know when it's ready to merge.

**Comment by TildeWill** [comment]
Should be good to go now that the tests have passed on Travis

**Comment by cantino** [comment]
My only concern is if people developing Agents will have trouble getting phantomjs/poltergeist working.  Is it pretty easy to install on Mac and Linux?

**Comment by TildeWill** [comment]
TravisCI has phantomjs installed by default. `npm install phantomjs` worked on my Linux environment (Cloud9), and `brew install phantomjs` has always been straight forward for me on a Mac. Is there a place in the README/Wiki about test writing/running that would benefit from added documentation?

**Comment by cantino** [comment]
There isn't, but I think there should be.  Would you be up for adding it, either to the README, or on the wiki?

**Comment by TildeWill** [comment]
Will do

**Comment by TildeWill** [comment]
Documentation added

File: Gemfile.lock [hunk 1]
```diff
@@ -648,4 +671,4 @@ DEPENDENCIES
   xmpp4r (~> 0.5.6)
 
 BUNDLED WITH
-   1.10.6
+   1.11.2
```
Comment (cantino) [inline Gemfile.lock]: Will this affect people with older Bundler versions?  I'm not entirely clear on how "BUNDLED WITH" works or what it does.

**Comment by cantino** [comment]
Looks great, thanks @TildeWill!

File: Gemfile.lock (hunk 1 above)
Comment (TildeWill) [inline Gemfile.lock]: Bundler spits out a warning when using an older version. Worst case scenario is that devs need to update their version of bundler to suppress the warning which is probably a good thing anyway.

**Pull request merged**